packages/stack/proxy/src/proxy.js

```diff
@@ -121,25 +126,40 @@ export class Proxy {
         const error = modifiedResp.respData.error.message || modifiedResp.respData.error;
         this.logger.error(__(`Error returned from the node: ${error}`));
         const rpcErrorObj = { "jsonrpc": "2.0", "error": { "code": -32603, "message": error }, "id": modifiedResp.respData.id };
-        return this.respondError(transport, rpcErrorObj, isWs);
+        return this.respondError(transport, rpcErrorObj);
       }
-      this.respondOK(transport, modifiedResp.respData, isWs);
+      this.respondOK(transport, modifiedResp.respData);
     }
     catch (resError) {
       // if was an error in response actions (resError), send the error in the response
       const error = resError.message || resError;
       this.logger.error(__(`Error executing response actions: ${error}`));
       const rpcErrorObj = { "jsonrpc": "2.0", "error": { "code": -32603, "message": error }, "id": modifiedRequest.reqData.id };
-      return this.respondError(transport, rpcErrorObj, isWs);
+      return this.respondError(transport, rpcErrorObj);
     }
   }
 
-  forwardRequestToNode(reqData) {
+  forwardRequestToNode(reqData, transport) {
     return new Promise((resolve, reject) => {
-      this.requestManager.send(reqData, (fwdReqErr, result) => {
+      const requestManager = new Web3RequestManager.Manager(this.endpoint);
+      // Watch from subscription data for events
+      if (this.isWs) {
+        requestManager.provider.on('data', (result, deprecatedResult) => {
```

Comment:
This goes a bit with what Pascal wrote below. Wouldn't listening to the `data` event on every request create a a "too many listeners" problem, and a memory problem too?

Reply:
There is only ever one listener per `RequestManager`. We could possibly try to figure out which requests coming in require a long-living connection (ie `eth_subscribe`), and for all others, use a single `RequestManager` instance. See my response to Pascal's comment below.

Reply:
We have gone with the suggestion above.